From 0790fb69c22010770c9c9787bbbb661addc8a447 Mon Sep 17 00:00:00 2001
From: ttimasdf <opensource@rabit.pw>
Date: Thu, 27 Nov 2025 21:41:31 +0800
Subject: [PATCH 3/4] cli7z: handle file open error for latest 7zip

---
 plugins/cli7zplugin/cliplugin.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/cli7zplugin/cliplugin.cpp b/plugins/cli7zplugin/cliplugin.cpp
index 41d2a1ad..1c585580 100644
--- a/plugins/cli7zplugin/cliplugin.cpp
+++ b/plugins/cli7zplugin/cliplugin.cpp
@@ -128,7 +128,7 @@ bool CliPlugin::readListLine(const QString &line)
     static const QLatin1String archiveInfoDelimiter2("----"); // 7z 9.04
     static const QLatin1String entryInfoDelimiter("----------");
 
-    if (line.startsWith(QLatin1String("Open ERROR: Can not open the file as [7z] archive"))) {
+    if (line.startsWith(QLatin1String("Open ERROR: Can not open the file as [7z] archive")) || line.endsWith(QLatin1String("Cannot open the file as archive"))) {
         Q_EMIT error(i18n("Listing the archive failed."));
         return false;
     }
-- 
2.51.2

