diff --git a/common/osdaction.cpp b/common/osdaction.cpp
index 78a24ea..ef75ddb 100644
--- a/common/osdaction.cpp
+++ b/common/osdaction.cpp
@@ -28,7 +28,7 @@ QList<OsdAction> OsdAction::availableActions()
         {SwitchToExternal, i18nd("kscreen_common", "Only use external screen"), QStringLiteral("osd-shutd-laptop")},
         {SwitchToInternal, i18nd("kscreen_common", "Only use built-in screen"), QStringLiteral("osd-shutd-screen")},
         {Clone, i18nd("kscreen_common", "Mirror screens"), QStringLiteral("osd-duplicate")},
-        {ExtendLeft, i18nd("kscreen_common", "Extend to left of built-in screen"), QStringLiteral("osd-sbs-left")},
+        {ExtendUp, i18nd("kscreen_common", "Extend above built-in screen"), QStringLiteral("view-split-top-bottom")},
         {ExtendRight, i18nd("kscreen_common", "Extend to right of built-in screen"), QStringLiteral("osd-sbs-sright")},
         {NoAction, i18nd("kscreen_common", "Leave unchanged"), QStringLiteral("dialog-cancel")},
     };
@@ -150,7 +150,7 @@ KScreen::SetConfigOperation *OsdAction::applyAction(const QSharedPointer<KScreen
         external->setPos({copy->logicalSizeForOutputInt(*internal).width(), 0});
         break;
     }
-    case KScreen::OsdAction::Action::ExtendLeft: {
+    case KScreen::OsdAction::Action::ExtendUp: {
         internal->setEnabled(true);
         external->setEnabled(true);
         external->setPos(QPoint());
@@ -167,7 +167,7 @@ KScreen::SetConfigOperation *OsdAction::applyAction(const QSharedPointer<KScreen
             currentMode = bestModeIt.value();
             external->setCurrentModeId(currentMode->id());
         }
-        internal->setPos({copy->logicalSizeForOutputInt(*external).width(), 0});
+        internal->setPos({0, copy->logicalSizeForOutputInt(*external).height()});
         break;
     }
     }
diff --git a/common/osdaction.h b/common/osdaction.h
index 1668fb4..f96c5e2 100644
--- a/common/osdaction.h
+++ b/common/osdaction.h
@@ -33,7 +33,7 @@ public:
         SwitchToExternal,
         SwitchToInternal,
         Clone,
-        ExtendLeft,
+        ExtendUp,
         ExtendRight,
     };
     Q_ENUM(Action)
diff --git a/tests/osdactions.cpp b/tests/osdactions.cpp
index 7e50185..30ca5c6 100644
--- a/tests/osdactions.cpp
+++ b/tests/osdactions.cpp
@@ -30,13 +30,13 @@ private Q_SLOTS:
     void testSwitchToExternal();
     void testSwitchToInternal();
     void testClone();
-    void testExtendLeft();
+    void testExtendUp();
     void testExtendRight();

     void testSwitchToExternalRotated();
     void testSwitchToInternalRotated();
     void testCloneRotated();
-    void testExtendLeftRotated();
+    void testExtendUpRotated();
     void testExtendRightRotated();

 private:
@@ -136,19 +136,19 @@ void TestOsdActions::testClone()
     QCOMPARE(external->pos(), QPoint(0, 0));
 }

-void TestOsdActions::testExtendLeft()
+void TestOsdActions::testExtendUp()
 {
     const KScreen::ConfigPtr currentConfig = loadConfig("laptopAndExternal.json");
     QVERIFY(currentConfig);

-    KScreen::SetConfigOperation *op = KScreen::OsdAction::applyAction(currentConfig, KScreen::OsdAction::Action::ExtendLeft);
+    KScreen::SetConfigOperation *op = KScreen::OsdAction::applyAction(currentConfig, KScreen::OsdAction::Action::ExtendUp);
     QVERIFY(op);
     QVERIFY(op->exec());
     auto [laptop, external] = findScreens(reloadConfig());

     QCOMPARE(laptop->isEnabled(), true);
-    // the display is not in the max vertical resolution, not the best horizontal
-    QCOMPARE(laptop->pos(), QPoint(1600, 0));
+    // the selected mode is 1600x1200
+    QCOMPARE(laptop->pos(), QPoint(0, 1200));
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
 }
@@ -220,19 +220,19 @@ void TestOsdActions::testCloneRotated()
     QCOMPARE(external->isEnabled(), false);
 }

-void TestOsdActions::testExtendLeftRotated()
+void TestOsdActions::testExtendUpRotated()
 {
     const KScreen::ConfigPtr currentConfig = loadConfig("steamdeckAndExternal.json");
     QVERIFY(currentConfig);

-    KScreen::SetConfigOperation *op = KScreen::OsdAction::applyAction(currentConfig, KScreen::OsdAction::Action::ExtendLeft);
+    KScreen::SetConfigOperation *op = KScreen::OsdAction::applyAction(currentConfig, KScreen::OsdAction::Action::ExtendUp);
     QVERIFY(op);
     QVERIFY(op->exec());
     auto [laptop, external] = findScreens(reloadConfig());

     QCOMPARE(laptop->isEnabled(), true);
-    // the display is not in the max vertical resolution, not the best horizontal
-    QCOMPARE(laptop->pos(), QPoint(1600, 0));
+    // the selected mode is 1600x1200
+    QCOMPARE(laptop->pos(), QPoint(0, 1200));
     QCOMPARE(laptop->rotation(), Output::Right);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
