From 70e2feb833f1fb349010aed1b0a7a2bbb4705a62 Mon Sep 17 00:00:00 2001
From: Misty <gyc990326@gmail.com>
Date: Fri, 10 Mar 2023 04:28:37 +0800
Subject: [PATCH 06/10] [mistypatch] feat: Use Escape to exit on Detached
 Windows

---
 .../main/java/docking/DetachedWindowNode.java | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/Ghidra/Framework/Docking/src/main/java/docking/DetachedWindowNode.java b/Ghidra/Framework/Docking/src/main/java/docking/DetachedWindowNode.java
index 635276c753..1f6e5fb71b 100644
--- a/Ghidra/Framework/Docking/src/main/java/docking/DetachedWindowNode.java
+++ b/Ghidra/Framework/Docking/src/main/java/docking/DetachedWindowNode.java
@@ -16,6 +16,8 @@
 package docking;
 
 import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.KeyEvent;
 import java.awt.event.WindowAdapter;
 import java.awt.event.WindowEvent;
 import java.util.*;
@@ -24,6 +26,7 @@ import java.util.Map.Entry;
 
 import javax.swing.*;
 
+import docking.actions.KeyBindingUtils;
 import org.jdom.Element;
 
 import generic.util.WindowUtilities;
@@ -324,6 +327,26 @@ class DetachedWindowNode extends WindowNode {
 		return finalTitles;
 	}
 
+	// Ghidra Mod - Esc2Exit - Escape to exit on DetachedWindowNode
+	private void installEscapeAction() {
+		JComponent pane = null;
+		if (window instanceof JDialog) {
+			pane = ((JDialog) window).getRootPane();
+		} else if (window instanceof JFrame) {
+			pane = ((JFrame) window).getRootPane();
+		}
+		Action escAction = new AbstractAction("ESCAPE") {
+			@Override
+			public void actionPerformed(ActionEvent ev) {
+				close();
+			}
+		};
+		var ESC_KEYSTROKE = KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0);
+		KeyBindingUtils.registerAction(pane, ESC_KEYSTROKE, escAction,
+				JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT);
+
+	}
+
 	private boolean isForceModal(JComponent comp) {
 		if (comp instanceof DockableComponent) {
 			comp = ((DockableComponent)comp).getProviderComponent();
@@ -349,6 +372,7 @@ class DetachedWindowNode extends WindowNode {
 			window = createFrame();
 		}
 
+		installEscapeAction();
 		if (dropTargetFactory != null) {
 			dropTargetHandler = dropTargetFactory.createDropTargetHandler(window);
 		}
-- 
2.51.0

