From 50fb750ced61a752d60a22cd21c4049f64620834 Mon Sep 17 00:00:00 2001
From: ttimasdf <opensource@rabit.pw>
Date: Sat, 28 Feb 2026 22:40:14 +0800
Subject: [PATCH 2/2] chore: remove linux root privilege check

---
 lib/main.dart | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/lib/main.dart b/lib/main.dart
index 9e3db56..1e9fc74 100644
--- a/lib/main.dart
+++ b/lib/main.dart
@@ -49,18 +49,6 @@ Future<void> _initializeApp() async {
     FileLogger().info('RustLib initialized');
     // initApp();
 
-    // Linux 下检测是否为 root 权限
-    if (!kIsWeb && Platform.isLinux) {
-      final env = Platform.environment;
-      if (env['USER'] != 'root' &&
-          env['SUDO_USER'] == null &&
-          env['UID'] != '0') {
-        FileLogger().error('Linux requires sudo privileges');
-        print('请使用 sudo 运行本程序！');
-        exit(1);
-      }
-    }
-
     WidgetsFlutterBinding.ensureInitialized();
 
     if (Platform.isMacOS) {
-- 
2.52.0

